commands/issue/update/issue_update.go

```diff
@@ -54,6 +54,12 @@ func NewCmdUpdate(f *cmdutils.Factory) *cobra.Command {
 			if m, _ := cmd.Flags().GetStringArray("unlabel"); len(m) != 0 {
 				l.RemoveLabels = gitlab.Labels(m)
 			}
+			if m, _ := cmd.Flags().GetBool("public"); m {
+				l.Confidential = gitlab.Bool(false)
+			}
+			if m, _ := cmd.Flags().GetBool("confidential"); m {
+				l.Confidential = gitlab.Bool(true)
+			}
```

Comment:
Instead of one overriding the other, there should be an error thrown if both flags are specified

```go
if cmd.Flags().Changed("confidential") && cmd.Flags().Changed("public") {
    return &cmdutil.FlagError{Err: errors.New("specify either of --public or --confidential")}
}
```